### PR TITLE
docs(readme): add pkg-docs link feat markdown enhancement

### PR DESCRIPTION
**Pull request by DanielHe4rt**
GoCQL README Page

---

Here's some small improvements to the `README.md` file, where I just managed to enhance some small point which was point to `pkg-go` docs. 

### Why I made this change?

I'm quite new to Golang and still learning about the environment yet. As expected from our Drivers, I tried to find our `Sphinx` Docs with more specifications with no luck. 

### Documentation Enhancements:

* Added a centered header and badges for build status, documentation, and protocol specs to improve the visual appeal and accessibility of important links.
* Introduced a table of contents to help users navigate through the document more easily.
* Reformatted section headers and added sub-sections for better clarity and organization. 
* Included a new "Contributing" section to guide users on how to contribute to the project.

### CODE REVIEW

**Comment by DanielHe4rt** [comment]
@annastuchlik @sylwiaszunejko please, review.

File: README.md [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
 
-![Build](https://github.com/scylladb/gocql/workflows/Build/badge.svg)
+![Build Passing](https://github.com/scylladb/gocql/workflows/Build/badge.svg)
+[![Read the Fork Driver Docs](https://img.shields.io/badge/Read_the_Docs-pkg_go-blue)](https://pkg.go.dev/github.com/scylladb/gocql#section-documentation)
+[![Protocol Specs](https://img.shields.io/badge/Protocol_Specs-ScyllaDB_Docs-blue)](https://github.com/scylladb/scylladb/blob/master/docs/dev/protocol-extensions.md)
+
+</div>
+
+<h1 align="center"> 
+  Scylla Shard-Aware Fork of gocql/gocql
```
Comment (dkropachev) [inline README.md:10]:
```suggestion
  Scylla Shard-Aware Fork of [apache/cassandra-gocql-driver](https://github.com/apache/cassandra-gocql-driver) 
```

File: README.md [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+</div>
+
+<h1 align="center"> 
+  Scylla Shard-Aware Fork of gocql/gocql 
+</h1>
+
+
+<img src="./.github/assets/logo.svg" width="200" align="left" />
 
 This is a fork of [gocql](https://github.com/gocql/gocql) package that we created at Scylla.
```
Comment (dkropachev) [inline README.md:16]:
```suggestion
This is a fork of [apache/cassandra-gocql-driver](https://github.com/apache/cassandra-gocql-driver) package that we created at Scylla.
```

File: README.md [hunk 3]
```diff
@@ -55,78 +74,82 @@ c := gocql.NewCluster(hosts...)
 // Enable token aware host selection policy, if using multi-dc cluster set a local DC.
 fallback := gocql.RoundRobinHostPolicy()
 if localDC != "" {
-	fallback = gocql.DCAwareRoundRobinPolicy(localDC)
+  fallback = gocql.DCAwareRoundRobinPolicy(localDC)
```
Comment (dkropachev) [inline README.md:77]: Nope, please revert

File: README.md [hunk 4]
```diff
@@ -55,78 +74,82 @@ c := gocql.NewCluster(hosts...)
...
   An example:
+
   ```go
   func (d *myDialer) DialContext(ctx context.Context, network, addr string) (net.Conn, error) {
       sourcePort := gocql.ScyllaGetSourcePort(ctx)
       localAddr, err := net.ResolveTCPAddr(network, fmt.Sprintf(":%d", sourcePort))
       if err != nil {
           return nil, err
       }
-	  d := &net.Dialer{LocalAddr: localAddr}
```
Comment (dkropachev) [inline README.md:92]: Should be on the same level as rest of the code

File: README.md [hunk 5]
```diff
@@ -55,78 +74,82 @@ c := gocql.NewCluster(hosts...)
...
 var results []int
 
 var id, value int
 for !iter.Scan(&id, &value) {
-	if id%2 == 0 {
-		results = append(results, value)
-	}
+  if id%2 == 0 {
+    results = append(results, value)
+  }
```
Comment (dkropachev) [inline README.md:152]: rest of the code indented with `\t` why woul you change it ?

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (DanielHe4rt) [reply README.md:152]: I'm following the base markdown [best practices](https://www.markdownguide.org/basic-syntax/). 

But if it's a problem I can rollback it.

Comment (dkropachev) [reply README.md:152]: It says:
```
Code Blocks
To create code blocks, indent every line of the block by at least four spaces or one tab.
```

So we can leave `\t`.

**Comment by DanielHe4rt** [comment]
@dkropachev did some soft reset without the identation changes.

**Review by dkropachev — APPROVED** [review]

**Review by sylwiaszunejko — APPROVED** [review]

**Pull request merged**